assets/js/frontend/course-video/video-blocks-manager.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * The Course Video Progression settings.
+ */
+const {
+	courseVideoRequired,
+	courseVideoAutoComplete,
+	courseVideoAutoPause,
+} = window.sensei.courseVideoSettings;
+
+/**
+ * Map of videos.
+ */
+const videos = {};
+
+/**
+ * Registers the videos to manage.
+ * @param {Object}   video
+ * @param {Function} video.pauseVideo              The function that pauses the video when invoked.
+ * @param {Function} video.registerVideoEndHandler Accepts a callback that is invoked when the video ends.
+ * @param {string}   video.url                     The source url of the video. Used as an id.
+ */
+export const registerVideo = ( {
+	pauseVideo = () => {},
+	registerVideoEndHandler = () => {},
+	url = '',
+} ) => {
+	if ( courseVideoRequired ) {
+		/**
+		 * Called when a required video for the current lesson is registered.
+		 *
+		 * @since 4.5.0
+		 *
+		 * @hook sensei.videoProgression.registerVideo Hook used to run an arbitrary code when new required
+		 *                                             video for the current lesson is registered.
+		 * @param {Object} video
+		 * @param {string} video.url The source url of the video.
+		 */
+		wp.hooks.doAction( 'sensei.videoProgression.registerVideo', { url } );
+		videos[ url ] = { pauseVideo, completed: false };
+		disableCompleteLessonButton();
+	}
+
+	registerVideoEndHandler( () => {
+		if ( courseVideoRequired ) {
+			/**
+			 * Called when a required video for the current lesson is finished playing.
+			 *
+			 * @since 4.5.0
+			 *
+			 * @hook sensei.videoProgression.videoEnded Hook used to run an arbitrary code when a required video
+			 *                                          for the current lesson is finished playing.
+			 * @param {Object} video
+			 * @param {string} video.url The source url of the video.
+			 */
+			wp.hooks.doAction( 'sensei.videoProgression.videoEnded', { url } );
+			videos[ url ].completed = true;
+			if ( areAllCompleted() ) {
+				enableCompleteLessonButton();
+			}
+		}
+
+		if ( courseVideoAutoComplete && areAllCompleted() ) {
+			submitCompleteLessonForm();
+		}
+	} );
+};
+
+/**
+ * Tells if all the required videos in the current page are completed.
+ *
+ * @return {boolean} True if all the videos are completed. False otherwise.
+ */
+const areAllCompleted = () => {
+	let allCompleted = true;
+	for ( const url in videos ) {
+		if ( ! videos[ url ].completed ) {
+			allCompleted = false;
+		}
+	}
+
+	/**
+	 * Tells if all the required videos for the current lesson are finished playing or not.
+	 *
+	 * @since 4.5.0
+	 *
+	 * @hook sensei.videoProgression.allCompleted Hook used to tell if all the required videos for the current lesson have finished playing.
+	 *
+	 * @param {boolean} allCompleted Whether all the required videos for the current lesson are completed.
+	 */
+	allCompleted = wp.hooks.applyFilters(
```

Comment:
I think that we need to import `@wordpress/hooks` as the package doesn't exist for lessons that don't use learning mode.

Reply:
Nice catch! Will fix. 👍

Reply:
Updated f11c8541e0bbb494c3fe6ec8c9871af7304ffca4